src/main/java/net/openhft/chronicle/map/monitor/MapCapacityMonitor.java

```diff
@@ -0,0 +1,50 @@
+package net.openhft.chronicle.map.monitor;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.map.ChronicleMap;
+import net.openhft.chronicle.threads.Threads;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.isNull;
+
+public class MapCapacityMonitor<K, V> implements Runnable{
+
+    public static final String MAP_CAPACITY_MONITOR_NAME = "map~capacity~monitor";
+
+    private static final boolean DISABLED = Jvm.getBoolean("chronicle.map.capacity.monitor.disable");
+    private int warnThresholdPercentage = Jvm.getInteger("chronicle.map.capacity.monitor.warn.percent", 80);
+    final ScheduledExecutorService executor;
+    final ChronicleMap<K, V> map;
+
+    public MapCapacityMonitor(ChronicleMap<K, V> map, int warnThresholdPercentage) {
```

Comment:
Suggestion: Add in a callback that can be provided by the user and called back when the threshold is breached, users can then plugin in their own handling logic, e.g.

```java
interface MapCapacityMonitorListener {
  void overThreshold(int usedSpacePercentage, int warnThreshold);
}
```

Reply:
Once a callback interface is in place it can be used to verify the behaviour in the test